### DISCUSSION TITLE
Node classification on multiple unconnected graphs (sets)

### DISCUSSION THREAD

**Discussion by twaclaw** [pyg-team/pytorch_geometric#7285] [Q&A]
Hi there,

I am interested in modeling the following problem but I am not sure how to proceed.

My dataset consists of a number of samples (~10k). **Each sample** is in turn a table of N combinations (rows) by M features and a 1 label (each table has dimensions [N,  M+1]). The idea of the classifier I want to build is to predict a class for each of the rows. The order of the rows in each table is not relevant.

I thought of modeling each sample as a subgraph. Each row in the table is a node with M features. The idea would then be to do node classification over multiple subgraphs. In other words, my dataset would have ~10k subgraphs, each with N nodes and M features. Given a single subgraph, I would like to predict the classes of the nodes or select one particular node.

1. Does this approach make sense?
2. I know it is possible to use the `DataLoader` to construct a large graph where the subgraphs are connected but what if the subgraphs are unconnected, i.e., what if they are sets?
3. Assuming the subgraphs are unconnected, is it possible to model this problem using `pyg`?
4. Does it makes sense to enforce connections between the nodes of each subgraph?

Many thanks in advance,

Diego.

**Comment by wsad1** [comment]
> Does this approach make sense?

Not completely. Because it looks like you don't have a graph, as in the rows don't have connections. Why do you want to use pyg  `DataLoader` in that case, instead using the native pytorch `DataLoader` makes more sense. 

Now you could create a graph by adding `edge_index` using [knn](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.KNNGraph.html#torch_geometric.transforms.KNNGraph) or [radius](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.RadiusGraph.html#torch_geometric.transforms.RadiusGraph). But I am not sure if thats the right approach for your problem.

**Reply by twaclaw** [reply]
Thanks, @wsad1. 

Are there `pyg` examples for sets? 

Best,
Diego

**Reply by wsad1** [reply]
I am not aware of such an example.  

> I know it is possible to use the DataLoader to construct a large graph where the subgraphs are connected but what if the subgraphs are unconnected, i.e., what if they are sets?

Also this shouldn't be a problem. The DataLoader should work on `Data` objects without `edge_index`.

**Reply by twaclaw** [reply]
Thanks.
> Also this shouldn't be a problem. The DataLoader should work on Data objects without edge_index

Yeah, but then I would have a single big graph instead of multiple small ones, wouldn't I?

**Reply by wsad1** [reply]
You can control that using the `batch_size` argument in [dataloader](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.DataLoader). If you have 10k samples and you set `batch_size` to say 32, the dataloader will return a single `Data` object with 32 samples combined.
